pkg/xds/topology/dataplanes.go

```diff
@@ -3,21 +3,39 @@ package topology
 import (
 	"context"
 
+	"github.com/pkg/errors"
+
+	"github.com/kumahq/kuma/pkg/core"
 	core_mesh "github.com/kumahq/kuma/pkg/core/resources/apis/mesh"
 	"github.com/kumahq/kuma/pkg/core/resources/manager"
 )
 
-// GetDataplanes returs list of Dataplane in provided Mesh and Ingresses (which are cluster-scoped, not mesh-scoped)
+// GetDataplanes returns list of Dataplane in provided Mesh and Ingresses (which are cluster-scoped, not mesh-scoped)
 func GetDataplanes(ctx context.Context, rm manager.ReadOnlyResourceManager, mesh string) (*core_mesh.DataplaneResourceList, error) {
 	dataplanes := &core_mesh.DataplaneResourceList{}
 	if err := rm.List(ctx, dataplanes); err != nil {
 		return nil, err
 	}
 	rv := &core_mesh.DataplaneResourceList{}
 	for _, d := range dataplanes.Items {
+		if err := ResolveAddress(d); err != nil {
+			return nil, err
+		}
 		if d.GetMeta().GetMesh() == mesh || d.Spec.IsIngress() {
 			_ = rv.AddItem(d)
 		}
 	}
 	return rv, nil
 }
+
+func ResolveAddress(dataplane *core_mesh.DataplaneResource) error {
```

Comment:
Overall, I need some explanation of why are we resolving an IP address. Please write a doc stating that EDS requires IP, not hostname but for some environments, it's much more convenient to have hostname in address.

Reply:
Do you mean doc as a comment right in the code? Do you think makes sense to mention ECS in this place in the code?

Reply:
Added info about reasons